packages/@aws-cdk/aws-ec2/lib/instance.ts

```diff
@@ -0,0 +1,263 @@
+
+  /**
+   * In which AZ to place the instance within the VPC
+   *
+   * @default - Random zone.
+   */
+  readonly availabilityZone?: string;
+
+  /**
+   * Whether the instance could initiate connections to anywhere by default
+   *
+   * @default true
+   */
+  readonly allowAllOutbound?: boolean;
+
+  /**
+   * The length of time to wait for the resourceSignalCount
+   *
+   * The maximum value is 43200 (12 hours).
+   *
+   * @default Duration.minutes(5)
+   */
+  readonly resourceSignalTimeout?: Duration;
+
+  /**
+   * VPC to launch the instance in.
+   */
+  readonly vpc: IVpc;
+
+  /**
+   * Type of instance to launch
+   */
+  readonly instanceType: InstanceType;
+
+  /**
+   * AMI to launch
+   */
+  readonly machineImage: IMachineImage;
+
+  /**
+   * Specific UserData to use
+   *
+   * The UserData may still be mutated after creation.
+   *
+   * @default - A UserData object appropriate for the MachineImage's
+   * Operating System is created.
+   */
+  readonly userData?: UserData;
+
+  /**
+   * An IAM role to associate with the instance profile assigned to this Auto Scaling Group.
+   *
+   * The role must be assumable by the service principal `ec2.amazonaws.com`:
+   *
+   * @example
+   *
+   *    const role = new iam.Role(this, 'MyRole', {
+   *      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com')
+   *    });
+   *
+   * @default A role will automatically be created, it can be accessed via the `role` property
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * The name of the instance
+   *
+   * @default - CDK generated name
+   */
+  readonly instanceName?: string;
+
+}
+
+export class Instance extends Resource implements IConnectable {
+
+  /**
+   * The type of OS the instance is running.
+   */
+  public readonly osType: OperatingSystemType;
+
+  /**
+   * Allows specify security group connections for the instance.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The IAM role assumed by the instance.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * UserData for the instance
+   */
+  public readonly userData: UserData;
+
+  /**
+   * the underlying instance resource
+   * @attribute
+   */
+  public readonly instance: CfnInstance;
+  /**
+   * @attribute
+   */
+  public readonly instanceId: string;
+  /**
+   * @attribute
+   */
+  public readonly instanceAvailabilityZone: string;
+  /**
+   * @attribute
+   */
+  public readonly instancePrivateDnsName: string;
+  /**
+   * @attribute
+   */
+  public readonly instancePrivateIp: string;
+  /**
+   * @attribute
+   */
+  public readonly instancePublicDnsName: string;
+  /**
+   * @attribute
+   */
+  public readonly instancePublicIp: string;
+
+  private readonly securityGroup: ISecurityGroup;
+  private readonly securityGroups: ISecurityGroup[] = [];
+
+  constructor(scope: Construct, id: string, props: InstanceProps) {
+    super(scope, id);
+
+    this.securityGroup = new SecurityGroup(this, 'InstanceSecurityGroup', {
```

Comment:
Instead of creating a security group here, this could be passed on props, then only create a default group is one is not supplied.
This gives the user ultimate flexibility.
In an enterprise or regulated space, the SecuityGroup generally would have been created upfront by another team.

Reply:
I like this idea because I do not like the SSH TCP access to the bastion server at all. So I will make the group suppliable and by default block any access. For the second part of your comment: I generally do not like the approach of other teams creating IAM roles or sec groups. The whole infrastructure should be defined and created by one team. If you want/have to have four-eye validation here I use the way of merge requests with approvals of the CISO/CSIRT or SCPs to prevent creating bad stuff. In an enterprise context, I would expect NACLs to be in place to prevent users from opening ports they are not allowed to.

Reply:
I see your point agree, but I work in environments that these rules apply, compliance forbids single team.
I have just created  PR for NACLs as they were missing.

Reply:
I see the compliance issue with single teams, but the question is if you can solve this on an org level instead of a code level. So by using four-eyes merges instead of separated AWS roles. But this is OT of this PR. Change is on the way.

Reply:
I am with @slipdexic - there are many users who simply are not allowed to create security groups or IAM roles and it makes sense to allow people to pass those in optionally.